frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+        isNightMode={isNightMode}
+        onNightModeChange={onNightModeChange}
+      />
+    ),
+    enabled: ({
+      isFullscreen,
+      dashboard,
+      hasNightModeToggle,
+      onNightModeChange,
+    }) =>
+      Boolean(
+        isFullscreen &&
+          !dashboard.archived &&
+          hasNightModeToggle &&
+          !!onNightModeChange,
+      ),
+  },
+  [BUTTON_CONFIG.FULLSCREEN_TOGGLE]: {
+    component: ({ isFullscreen, onFullscreenChange }) => (
+      <FullscreenToggle
+        isFullscreen={isFullscreen}
+        onFullscreenChange={onFullscreenChange}
+      />
+    ),
+    enabled: ({ isFullscreen, isEmpty, isPublic }) =>
+      !isEmpty && (isPublic || isFullscreen),
+  },
+  [BUTTON_CONFIG.DASHBOARD_HEADER_ACTION_DIVIDER]: {
+    component: () => <DashboardHeaderActionDivider />,
+    enabled: ({ isEditing }) => !isEditing,
+  },
+  [BUTTON_CONFIG.DASHBOARD_BOOKMARK]: {
+    component: () => <DashboardBookmark />,
+    enabled: ({ isEditing, dashboard }) => !isEditing && !dashboard.archived,
+  },
+  [BUTTON_CONFIG.DASHBOARD_INFO]: {
+    component: () => <DashboardInfoButton />,
+    enabled: ({ isEditing }) => !isEditing,
+  },
+  [BUTTON_CONFIG.DASHBOARD_ACTION_MENU]: {
+    component: ({
+      onFullscreenChange,
+      isFullscreen,
+      dashboard,
+      canEdit,
+      location,
+    }) => (
+      <DashboardActionMenu
+        items={getExtraButtons({
+          onFullscreenChange,
+          isFullscreen,
+          dashboard,
+          canEdit,
+          pathname: location.pathname,
+        })}
+      />
+    ),
+    enabled: ({ isFullscreen, isEditing, isAnalyticsDashboard, dashboard }) =>
+      !isFullscreen &&
+      !isEditing &&
+      !isAnalyticsDashboard &&
+      !dashboard.archived,
+  },
+  [BUTTON_CONFIG.FULLSCREEN_ANALYTICS_DASHBOARD]: {
+    component: ({ isFullscreen, onFullscreenChange }) => (
+      <FullscreenAnalyticsDashboard
+        isFullscreen={isFullscreen}
+        onFullscreenChange={onFullscreenChange}
+      />
+    ),
+    enabled: ({ isAnalyticsDashboard }) => isAnalyticsDashboard,
+  },
+};
+
+export const DashboardHeaderButtonRow = withRouter(
+  ({
+    buttonKeys = null,
+    isPublic = false,
+    isEmpty = false,
+    ...props
+  }: {
+    buttonKeys?: ButtonKey[] | null;
+  } & DashboardHeaderButtonRowProps) => {
+    const formInput = useSelector(getPulseFormInput);
+    const isAdmin = useSelector(getUserIsAdmin);
+    const canManageSubscriptions = useSelector(canManageSubscriptionsSelector);
+
+    const dashboard = useSelector(getDashboardComplete);
+    const canEdit = Boolean(dashboard?.can_write && !dashboard?.archived);
+    const isAnalyticsDashboard = isInstanceAnalyticsCollection(
+      props.collection,
+    );
+
+    const hasModelActionsEnabled = useSelector(getHasModelActionsEnabled);
+
+    const isEditing = useSelector(getIsEditing);
+
+    const visibleKeys: ButtonKey[] =
+      buttonKeys || (Object.keys(buttonConfigs) as ButtonKey[]);
```

Comment:
I'd set `Object.keys(buttonConfigs)` as a default value to `buttonKeys` I was a bit confused why we don't need to pass `buttonKeys` and it still works here

https://github.com/metabase/metabase/blob/e98992501909e65149f35cd2837f43a131d404d2/frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderView.tsx#L94-L103